### PR TITLE
Fix: FollowButton state management and minor ui bugs

### PR DESCRIPTION
**Pull request by Shreyassp002**
This PR addresses issues where the **Follow button** state was not updating correctly across the ListenScreen and BaseProfileScreen also some ui bugs fixes. The following changes have been made:
- Corrected the styling of the **Follow Button** to match the light theme.
- Fixed text color inconsistencies in the **Compatibility** section for better visual consistency.
- Fixed **Follow Button** state management across ListenScreen and BaseProfileScreen.

### Issue:
issue 1

issue 2

issue 3

https://github.com/user-attachments/assets/66407a61-a485-4833-a6c3-215b8bac6b77

### Fixes:
Screenshot_2024-12-15-23-36-40-908_org listenbrainz android debug

Screenshot_2024-12-15-23-37-03-659_org listenbrainz android debug

Screenshot_2024-12-15-23-37-54-849_org listenbrainz android debug

https://github.com/user-attachments/assets/d1d79361-85cb-48c6-b496-641e08ab50e0

Let me know if there are any changes needed to the color scheme or any other adjustments!

### CODE REVIEW

**Comment by 07jasjeet** [comment]
Hi, thanks for the PR. We would like to keep the follow button same throughout the app, if you could replace the button with existing component entirely, it would be really great!

**Comment by Shreyassp002** [comment]
Hi @07jasjeet ,

Thank you for the feedback! Just to ensure consistency:

1. Could you confirm which existing button component you'd like me to replace the Follow button with? If possible, could you point me to the file or component name where it's implemented?

2. Regarding the state management updates I made for the Follow button across the ListenScreen and BaseProfileScreen, should I revert those changes as well, or should I adapt them to work with the existing component?

**Comment by 07jasjeet** [comment]
1. @Shreyassp002 [this](https://github.com/metabrainz/listenbrainz-android/blob/main/app/src/main/java/org/listenbrainz/android/ui/components/FollowButton.kt) is the component I'm talking about. 
2. State management changes look good, no need to revert.

**Comment by Shreyassp002** [comment]
Hi @07jasjeet ,

I’ve updated the ListensScreen to use the requested component. The BaseProfileScreen already had the correct component in place, so no changes were needed there.

photo_2024-12-24_17-52-27
photo_2024-12-24_17-52-31


Please let me know if there is anything else that needs attention or further adjustments.

File: app/src/main/java/org/listenbrainz/android/ui/screens/profile/listens/ListensScreen.kt [hunk 1]
```diff
@@ -734,29 +739,19 @@ private fun FollowCard(username: String?, onFollowButtonClick: (String?, Boolean
...
-                        false -> "Follow"
-                    }, color = when(followStatus){
-                        true -> ListenBrainzTheme.colorScheme.followerCardTextColor
-                        false -> Color.White
+            Box(){
+                FollowButton(
+                    modifier = Modifier,
+                    isFollowedState = followStatus,
+                    buttonColor = lb_purple,
+                    followedStateTextColor = ListenBrainzTheme.colorScheme.text,
```
Comment (07jasjeet) [inline app/src/main/java/org/listenbrainz/android/ui/screens/profile/listens/ListensScreen.kt:747]:
```suggestion
                    followedStateTextColor = ListenBrainzTheme.colorScheme.lbSignature,
```

**Review by 07jasjeet — CHANGES_REQUESTED** [review]
Last bit changes

**Review by 07jasjeet — APPROVED** [review]
LGTM, Thanks for fixing this UI issue!

**Pull request merged**